Release 2.1.8 introduces exception in macro expansion.

We see several projects where the ZIO `2.1.8` update builds are failing.

Test code does not compile. Error reported:
```
exception during macro expansion: 
[error] java.lang.Throwable: This can't happen
[error] 	at zio.internal.macros.Graph.$anonfun$build$1(Graph.scala:147)
[error] 	at zio.internal.macros.Graph.build(Graph.scala:189)
[error] 	at zio.internal.macros.Graph.$anonfun$buildNodes$5(Graph.scala:29)
[error] 	at zio.internal.macros.Graph.$anonfun$buildNodes$2(Graph.scala:28)
[error] 	at zio.internal.macros.Graph.buildNodes(Graph.scala:27)
[error] 	at zio.internal.macros.LayerBuilder.build(LayerBuilder.scala:93)
[error] 	at zio.internal.macros.LayerMacroUtils.constructLayer(LayerMacroUtils.scala:131)
[error] 	at zio.internal.macros.LayerMacroUtils.constructLayer$(LayerMacroUtils.scala:41)
[error] 	at zio.internal.macros.LayerMacros.constructLayer(LayerMacros.scala:9)
[error] 	at zio.internal.macros.LayerMacroUtils.provideBaseImpl(LayerMacroUtils.scala:139)
[error] 	at zio.internal.macros.LayerMacroUtils.provideBaseImpl$(LayerMacroUtils.scala:134)
[error] 	at zio.internal.macros.LayerMacros.provideBaseImpl(LayerMacros.scala:9)
[error] 	at zio.internal.macros.LayerMacros.provideSomeImpl(LayerMacros.scala:36)
[error]     ).provideSome(TestImageConsumer.layer) @@ TestAspect.before(emptyLocalFiles() *> clearRemote()) @@
[error]                  ^
```

I don't know if this is related, but we also have some projects where the running of `test` fails for the `2.1.8` update branch.
So after entering `test` in sbt we see the error:
```
[error] Exception in thread "zio-fiber-1834380461,394448945,558798678,1009396526" java.lang.NoSuchMethodError: 'zio.ZIO zio.ZIO$.$anonfun$uninterruptible$1(scala.Function0, java.lang.Object)'
[error] 	at zio.test.results.ResultFileOps$Json$.apply(ResultFileOps.scala:82)
[error] 	at zio.test.results.ResultFileOps$.$anonfun$live$1(ResultFileOps.scala:15)
[error] 	at zio.ZLayer$ScopedPartiallyApplied$.$anonfun$apply$extension$1(ZLayer.scala:2027)
[error] 	at zio.ZLayer$ScopedEnvironmentPartiallyApplied$.$anonfun$apply$extension$2(ZLayer.scala:2034)
[error] 	at zio.ZLayer.$anonfun$scope$17(ZLayer.scala:431)
[error] 	at zio.ZLayer$MemoMap$$anon$25.$anonfun$getOrElseMemoize$17(ZLayer.scala:1986)
[error] Exception in thread "zio-fiber-734856381,1009396526" java.lang.InterruptedException: Interrupted by thread "zio-fiber-1009396526"
[error] 	Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1009396526"
[error] 		Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1009396526"
[error] 			Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1009396526"
[error] 				Suppressed: java.lang.InterruptedException: Interrupted by thread "zio-fiber-1009396526"
[error] Caused by: java.lang.NoSuchMethodError: 'zio.ZIO zio.ZIO$.$anonfun$uninterruptible$1(scala.Function0, java.lang.Object)'
[error] 	at zio.test.results.ResultFileOps$Json$.apply(ResultFileOps.scala:82)
[error] 	at zio.test.results.ResultFileOps$.$anonfun$live$1(ResultFileOps.scala:15)
[error] 	at zio.ZLayer$ScopedPartiallyApplied$.$anonfun$apply$extension$1(ZLayer.scala:2027)
[error] 	at zio.ZLayer$ScopedEnvironmentPartiallyApplied$.$anonfun$apply$extension$2(ZLayer.scala:2034)
[error] 	at zio.ZLayer.$anonfun$scope$17(ZLayer.scala:431)
[error] 	at zio.ZLayer$MemoMap$$anon$25.$anonfun$getOrElseMemoize$17(ZLayer.scala:1986)
[error] 	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1077)
[error] 	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1162)
[error] 	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1097)
[error] 	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
[error] 	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1067)
[error] 	at zio.internal.FiberRuntime.runLoop(FiberRuntime.scala:1162)
[error] 	at zio.internal.FiberRuntime.evaluateEffect(FiberRuntime.scala:413)
[error] 	at zio.internal.FiberRuntime.evaluateMessageWhileSuspended(FiberRuntime.scala:489)
[error] 	at zio.internal.FiberRuntime.drainQueueOnCurrentThread(FiberRuntime.scala:250)
[error] 	at zio.internal.FiberRuntime.run(FiberRuntime.scala:138)
[error] 	at zio.internal.ZScheduler$$anon$3.run(ZScheduler.scala:380)

```

--- a/core-tests/shared/src/test/scala/zio/ZLayerSpec.scala
+++ b/core-tests/shared/src/test/scala/zio/ZLayerSpec.scala
@@ -348,10 +348,11 @@ object ZLayerSpec extends ZIOBaseSpec {
         }
 
         for {
-          ref    <- makeRef
-          layer1  = makeLayer1(ref).fresh.map(identity)
-          layer2  = makeLayer2(ref)
-          _      <- (layer1 ++ (layer1 >>> layer2)).build
+          ref   <- makeRef
+          layer1 = makeLayer1(ref)
+          layer2 = makeLayer2(ref)
+          _ <- (ZIO.service[Service1] *> ZIO.service[Service2])
+                 .provide(layer1.fresh.map(identity).update(a => a), layer2)
           result <- ref.get
         } yield assert(result)(hasSameElements(Vector("1", "1", "2")))
       } @@ exceptJS(nonFlaky),

--- a/core-tests/shared/src/test/scala/zio/autowire/AutoWireSpec.scala
+++ b/core-tests/shared/src/test/scala/zio/autowire/AutoWireSpec.scala
@@ -2,7 +2,7 @@ package zio.autowire
 
 import zio._
 import zio.internal.macros.StringUtils.StringOps
-import zio.test.Assertion._
+import zio.test.Assertion.{anything, equalTo, isLeft}
 import zio.test._
 import scala.annotation.nowarn
 
@@ -132,25 +132,6 @@ object AutoWireSpec extends ZIOBaseSpec {
             val provided = ZIO.service[Int].provideLayer(layer)
             assertZIO(provided)(equalTo(128))
           },
-          test("building fresh layer only once") {
-            var timesBuilt = 0
-
-            trait Service1
-            object Service1 {
-              val live: ZLayer[Any, Throwable, Service1] =
-                ZLayer.fromZIO(ZIO.attempt(timesBuilt += 1).as(new Service1 {}))
-            }
-
-            trait Service2
-            object Service2 {
-              val live: ZLayer[Service1, Nothing, Service2] =
-                ZLayer.fromZIO(ZIO.unit.as(new Service2 {}))
-            }
-
-            assertZIO(
-              ZLayer.make[Service1 & Service2](Service1.live.fresh, Service2.live).build *> ZIO.succeed(timesBuilt)
-            )(equalTo(1))
-          },
           test("correctly decomposes nested, aliased intersection types") {
             type StringAlias           = String
             type HasBooleanDoubleAlias = Boolean with Double
@@ -250,98 +231,10 @@ object AutoWireSpec extends ZIOBaseSpec {
                 ZLayer.succeed(true) ++ ZLayer.succeed(100.1) >>> layer
               )
             assertZIO(provided)(equalTo(128))
-          },
-          test("makeSome 1 layer") {
-            val test1 = typeCheck {
-              """def test1[A, B](a: ZLayer[A, Nothing, B]): ZLayer[A, Nothing, B] =
-              ZLayer.makeSome[A, B](a)"""
-            }
-
-            val test2 = typeCheck {
-              """def test2[A](a: ZLayer[A, Nothing, Int]): ZLayer[A, Nothing, A & Int] =
-              ZLayer.makeSome[A, A & Int](a)"""
-            }
-
-            assertZIO(test1)(isRight(anything)) &&
-            assertZIO(test2)(isRight(anything))
-          },
-          test("makeSome 2 simple layers") {
-
-            def test1[I1, O1, I2](
-              a: ZLayer[I1, Nothing, O1],
-              b: ZLayer[I2, Nothing, Int]
-            ): ZLayer[I1 & I2, Nothing, O1 & Int] =
-              ZLayer.makeSome[I1 & I2, O1 & Int](a, b)
-
-            def test2[I1, O1](a: ZLayer[I1, Nothing, O1], b: ZLayer[O1, Nothing, Int]): ZLayer[I1, Nothing, O1 & Int] =
-              ZLayer.makeSome[I1, O1 & Int](a, b)
-
-            def test3[I1, O1](a: ZLayer[I1, Nothing, O1], b: ZLayer[I1, Nothing, Int]): ZLayer[I1, Nothing, O1 & Int] =
-              ZLayer.makeSome[I1, O1 & Int](a, b)
-
-            val t1 = test1 _
-            val t2 = test2 _
-            val t3 = test3 _
-
-            assertTrue(dummy(t1, t2, t3))
-
-          },
-          test("makeSome 2 complex layers") {
-
-            def test1[R, R1](a: ZLayer[R1 & Int, Nothing, R], b: ZLayer[Int, Nothing, R1]): ZLayer[Int, Nothing, R] =
-              ZLayer.makeSome[Int, R](a, b)
-
-            def test2[I1, I3, O1, O2](
-              a: ZLayer[I1 & Double, Nothing, O1 & O2],
-              b: ZLayer[I3 & Float, Nothing, Int]
-            ): ZLayer[I1 & Double & I3 & Float, Nothing, O1 & O2 & Int] =
-              ZLayer.makeSome[I1 & Double & I3 & Float, O1 & O2 & Int](a, b)
-
-            def test3[I1, I4, O1, O2](
-              a: ZLayer[I1 & Double, Nothing, O1 & O2],
-              b: ZLayer[I1 & Float, Nothing, Int]
-            ): ZLayer[I1 & Double & Float, Nothing, O1 & O2 & Int] =
-              ZLayer.makeSome[I1 & Double & Float, O1 & O2 & Int](a, b)
-
-            def test4[I1, O1](
-              a: ZLayer[I1 & Double, Nothing, O1 & String],
-              b: ZLayer[O1 & Float, Nothing, Int]
-            ): ZLayer[I1 & Double & Float, Nothing, O1 & String & Int] =
-              ZLayer.makeSome[I1 & Double & Float, O1 & String & Int](a, b)
-
-            val t1 = test1 _
-            val t2 = test2 _
-            val t3 = test3 _
-            val t4 = test4 _
-
-            assertTrue(dummy(t1, t2, t3, t4))
-          },
-          test("makeSome complex layer and providing") {
-
-            def test1[R, R1](a: ZLayer[R1 & Int, Nothing, R], b: ZLayer[Int, Nothing, R1]): ZLayer[Int, Nothing, R] =
-              ZLayer.makeSome[Int, R](a, b)
-
-            val la = ZLayer {
-              (ZIO.service[String] <*> ZIO.service[Int]).map { case (str, int) =>
-                (str.length + int).toLong
-              }
-            }
-            val lb = ZLayer(ZIO.service[Int].map(n => n.toString))
-
-            val program =
-              ZIO.service[Long].provideLayer(ZLayer.succeed(8) >>> test1(la, lb))
-
-            assertZIO(program)(equalTo(9.toLong))
           }
         )
       )
     )
-
-  def dummy(f: Any*): Boolean = {
-    var l: List[Any] = f.toList
-    l = Nil
-    l.isEmpty
-  }
 
   object TestLayer {
     trait OldLady {

--- a/core/shared/src/main/scala-2/zio/internal/macros/LayerMacroUtils.scala
+++ b/core/shared/src/main/scala-2/zio/internal/macros/LayerMacroUtils.scala
@@ -160,7 +160,7 @@ private[zio] trait LayerMacroUtils {
 
     intersectionTypes
       .map(_.dealias)
-      .filterNot(t => typeOf[Any] <:< t)
+      .filterNot(_.isAny)
       .distinct
   }
 

--- a/core/shared/src/main/scala/zio/internal/macros/Graph.scala
+++ b/core/shared/src/main/scala/zio/internal/macros/Graph.scala
@@ -2,175 +2,50 @@ package zio.internal.macros
 
 import zio.internal.macros.LayerTree._
 
-final case class Graph[Key, A](
-  nodes: List[Node[Key, A]],
-  keyEquals: (Key, Key) => Boolean,
-  environment: Key => Node[Key, A],
-  envKeys: List[Key]
-) {
+final case class Graph[Key, A](nodes: List[Node[Key, A]], keyEquals: (Key, Key) => Boolean) {
 
-  // Map assigning to each type the times that it must be built
-  // -1 designs a `Key` from the environment
-  private var neededKeys: Map[Key, Int] = Map.empty
-  // Dependencies to pass to next iteration of buildComplete
-  private var dependencies: List[Key]    = Nil
-  private var envDependencies: List[Key] = Nil
-
-  private var usedEnvKeys: Set[Key] = Set.empty
-  def usedRemainders(): Set[A]      = usedEnvKeys.map(environment(_)).map(_.value)
-  private var topLevel              = true
-
-  def buildNodes(
-    outputs: List[Key],
-    sideEffectNodes: List[Node[Key, A]]
-  ): Either[::[GraphError[Key, A]], LayerTree[A]] = for {
-    _           <- mkNeededKeys(outputs ++ sideEffectNodes.flatMap(_.inputs), true)
-    sideEffects <- forEach(sideEffectNodes)(buildNode).map(_.combineHorizontally)
-    rightTree   <- build(outputs).map(_._1)
-    leftTree    <- buildComplete(constructDeps())
-  } yield leftTree >>> (rightTree ++ sideEffects)
-
-  private def buildComplete(outputs: List[Key]): Either[::[GraphError[Key, A]], LayerTree[A]] =
-    if (!outputs.isEmpty)
-      for {
-        _         <- Right(restartKeys())
-        _         <- mkNeededKeys(outputs)
-        rightTree <- build(outputs).map(_._1)
-        leftTree  <- buildComplete(constructDeps())
-      } yield leftTree >>> rightTree
-    else Right(LayerTree.empty)
-
-  private def constructDeps(): List[Key] =
-    if (dependencies.isEmpty) dependencies
-    else distinctKeys(dependencies) ++ distinctKeys(envDependencies)
-
-  /**
-   * Restarts variables for next iteration of buildComplete
-   */
-  private def restartKeys(): Unit = {
-    neededKeys = Map.empty
-    dependencies = Nil
-    envDependencies = Nil
-  }
-
-  private def distinctKeys(keys: List[Key]): List[Key] = {
-    var distinct: List[Key] = List.empty
-    for (k <- keys) {
-      if (!distinct.exists(k2 => keyEquals(k, k2) || keyEquals(k2, k))) distinct = k :: distinct
+  def buildComplete(outputs: List[Key]): Either[::[GraphError[Key, A]], LayerTree[A]] =
+    forEach(outputs) { output =>
+      getNodeWithOutput[GraphError[Key, A]](output, error = GraphError.MissingTopLevelDependency(output))
+        .flatMap(node => buildNode(node, Set(node)))
     }
-    distinct.reverse
-  }
+      .map(_.distinct.combineHorizontally)
 
-  /**
-   * Initializes neededKeys
-   */
-  def mkNeededKeys(
-    outputs: List[Key],
-    topLevel: Boolean = false,
-    seen: Set[Node[Key, A]] = Set.empty,
-    parent: Option[Node[Key, A]] = None
-  ): Either[::[GraphError[Key, A]], Unit] = {
-    var created: List[Key]          = List.empty
-    var (envOutputs, normalOutputs) = outputs.partition(isEnv(_))
-
-    envOutputs.map(addEnv(_))
-
-    forEach(normalOutputs) { output =>
-      if (created.exists(k => keyEquals(k, output))) {
-        if (neededKeys.get(output).isEmpty) throw new Throwable("This can't happen.")
-        Right(())
-      } else {
-        for {
-          node <- getNodeWithOutput[GraphError[Key, A]](
-                    output,
-                    error =
-                      if (topLevel || parent.isEmpty) Some(GraphError.MissingTopLevelDependency(output))
-                      else Some(GraphError.missingTransitiveDependency(parent.get, output))
-                  )
-          nodeOutputs = node.outputs.map(out => findKey(out, outputs))
-          _          <- Right(nodeOutputs.map(addKey(_)))
-          _          <- Right { created = nodeOutputs ++ created }
-          _ <- parent match {
-                 case Some(p) => assertNonCircularDependency(p, seen, node)
-                 case None    => Right(())
-               }
-          _ <- mkNeededKeys(node.inputs, false, seen + node, Some(node))
-        } yield ()
-      }
-    }.map(_ => ())
-  }
-
-  private def findKey(key: Key, keys: List[Key]): Key =
-    keys.find(k => keyEquals(key, k)).getOrElse(key)
-
-  private def addEnv(key: Key): Unit = {
-    val keyFromEnv = envKeys
-      .find(env => keyEquals(key, env) || keyEquals(env, key))
-      .getOrElse(throw new Throwable("This shouldn't happen"))
-    usedEnvKeys = usedEnvKeys + keyFromEnv
-    neededKeys.get(key) match {
-      case Some(_) => ()
-      case None    => neededKeys = neededKeys + (key -> -1)
-    }
-  }
-  private def addKey(key: Key): Unit =
-    neededKeys.get(key) match {
-      case Some(-1) => ()
-      case Some(n)  => neededKeys = neededKeys + (key -> (n + 1))
-      case None     => neededKeys = neededKeys + (key -> 1)
-    }
+  def buildNodes(nodes: List[Node[Key, A]]): Either[::[GraphError[Key, A]], LayerTree[A]] =
+    forEach(nodes)(buildNode).map(_.combineHorizontally)
 
   private def buildNode(node: Node[Key, A]): Either[::[GraphError[Key, A]], LayerTree[A]] =
-    build(node.inputs).map { case (deps, allEnv) =>
-      if (allEnv) LayerTree.succeed(node.value)
-      else deps >>> LayerTree.succeed(node.value)
+    forEach(node.inputs) { output =>
+      getNodeWithOutput[GraphError[Key, A]](output, error = GraphError.missingTransitiveDependency(node, output))
+        .flatMap(node => buildNode(node, Set(node)))
     }
-
-  private def tapKey(key: Key): Option[Int] =
-    neededKeys.get(key) match {
-      case Some(n) => Some(n)
-      case None    => neededKeys.find(pair => keyEquals(pair._1, key) || keyEquals(key, pair._1)).map(_._2)
-    }
-
-  /**
-   * Builds a layer containing only types that appears once. Types appearing
-   * more than once are replaced with ZLayer.environment[_] and left for the
-   * next iteration of buildComplete to create.
-   */
-  private def build(outputs: List[Key]): Either[::[GraphError[Key, A]], (LayerTree[A], Boolean)] =
-    forEach(outputs) { output =>
-      if (isEnv(output)) {
-        envDependencies = output :: envDependencies
-        Right((LayerTree.succeed(environment(output).value), true))
-      } else
-        tapKey(output) match {
-          case None => throw new Throwable(s"This can't happen")
-          case Some(1) =>
-            getNodeWithOutput[GraphError[Key, A]](output).flatMap(node => buildNode(node).map(tree => (tree, false)))
-          case Some(n) => {
-            dependencies = output :: dependencies
-            Right((LayerTree.succeed(environment(output).value), true))
-          }
-        }
-    }.map { deps =>
-      (deps.map(_._1).distinct.combineHorizontally, deps.forall(_._2))
-    }
+      .map(_.distinct.combineHorizontally)
+      .map(_ >>> LayerTree.succeed(node.value))
 
   def map[B](f: A => B): Graph[Key, B] =
-    Graph(nodes.map(_.map(f)), keyEquals, key => environment(key).map(f), envKeys)
+    Graph(nodes.map(_.map(f)), keyEquals)
 
   private val nodeWithOutputCache = new java.util.HashMap[Key, Option[Node[Key, A]]]
 
-  private def getNodeWithOutput[E](output: Key, error: Option[E] = None): Either[::[E], Node[Key, A]] =
-    nodeWithOutputCache
-      .computeIfAbsent(output, findNodeWithOutput)
-      .toRight(error.map(e => ::(e, Nil)).getOrElse(throw new Throwable("This can't happen")))
+  private def getNodeWithOutput[E](output: Key, error: => E): Either[::[E], Node[Key, A]] =
+    nodeWithOutputCache.computeIfAbsent(output, findNodeWithOutput).toRight(::(error, Nil))
 
   private def findNodeWithOutput(output: Key): Option[Node[Key, A]] =
     nodes.find(_.outputs.exists(keyEquals(_, output)))
 
-  private def isEnv(key: Key): Boolean =
-    envKeys.exists(env => keyEquals(key, env) || keyEquals(env, key))
+  private def buildNode(
+    node: Node[Key, A],
+    seen: Set[Node[Key, A]]
+  ): Either[::[GraphError[Key, A]], LayerTree[A]] =
+    forEach(node.inputs) { input =>
+      for {
+        out    <- getNodeWithOutput(input, error = GraphError.missingTransitiveDependency(node, input))
+        _      <- assertNonCircularDependency(node, seen, out)
+        result <- buildNode(out, seen + out)
+      } yield result
+    }.map {
+      _.distinct.combineHorizontally >>> LayerTree.succeed(node.value)
+    }
 
   private def assertNonCircularDependency(
     node: Node[Key, A],

--- a/core/shared/src/main/scala/zio/internal/macros/LayerBuilder.scala
+++ b/core/shared/src/main/scala/zio/internal/macros/LayerBuilder.scala
@@ -87,17 +87,22 @@ final case class LayerBuilder[Type, Expr](
      * constructed ZLayer that will build the target types. This, of course, may
      * fail with one or more GraphErrors.
      */
-    val nodes: List[Node[Type, Expr]] = providedLayerNodes
-    val graph                         = Graph(nodes, typeEquals, typeToNode, remainder)
-    val layerTreeEither: Either[::[GraphError[Type, Expr]], LayerTree[Expr]] =
-      graph.buildNodes(target.filterNot(typeEquals(_, sideEffectType)), sideEffectNodes)
+    val layerTreeEither: Either[::[GraphError[Type, Expr]], LayerTree[Expr]] = {
+      val nodes: List[Node[Type, Expr]] = providedLayerNodes ++ remainderNodes ++ sideEffectNodes
+      val graph                         = Graph(nodes, typeEquals)
+
+      for {
+        original    <- graph.buildComplete(target)
+        sideEffects <- graph.buildNodes(sideEffectNodes)
+      } yield sideEffects ++ original
+    }
 
     layerTreeEither match {
       case Left(buildErrors) =>
         reportBuildErrors(buildErrors)
 
       case Right(tree) =>
-        warnUnused(tree, graph.usedRemainders())
+        warnUnused(tree)
         maybeDebug.foreach(debugLayer(_, tree))
         foldTree(tree)
     }
@@ -144,7 +149,7 @@ final case class LayerBuilder[Type, Expr](
    * used. This will also warn about any specified remainders that aren't
    * actually required, in the case of provideSome/provideCustom.
    */
-  private def warnUnused(tree: LayerTree[Expr], usedRemainders: Set[Expr]): Unit = {
+  private def warnUnused(tree: LayerTree[Expr]): Unit = {
     val usedLayers =
       tree.map(showExpr).toSet
 
@@ -156,8 +161,7 @@ final case class LayerBuilder[Type, Expr](
       reportWarn(message)
     }
 
-    val unusedRemainderLayers =
-      remainderNodes.filterNot(node => usedRemainders.map(showExpr(_)).apply(showExpr(node.value)))
+    val unusedRemainderLayers = remainderNodes.filterNot(node => usedLayers(showExpr(node.value)))
 
     method match {
       case ProvideMethod.Provide => ()

--- a/docs/reference/service-pattern/service-pattern.md
+++ b/docs/reference/service-pattern/service-pattern.md
@@ -182,14 +182,14 @@ Similarly, we need to implement the `BlobStorage` and `MetadataRepo` services:
 
 ```scala mdoc:silent
 object InmemoryBlobStorage {
-  val layer: ZLayer[Any, Nothing, BlobStorage] = 
+  val layer = 
     ZLayer {
       ???
     } 
 }
 
 object InmemoryMetadataRepo {
-  val layer: ZLayer[Any, Nothing, MetadataRepo] = 
+  val layer = 
     ZLayer {
       ???
     }

--- a/docs/reference/service-pattern/the-three-laws-of-zio-environment.md
+++ b/docs/reference/service-pattern/the-three-laws-of-zio-environment.md
@@ -129,7 +129,7 @@ object DocRepo {
 
 
 object InmemoryBlobStorage {
-  val layer: ZLayer[Any, Nothing, BlobStorage] = 
+  val layer = 
     ZLayer {
       ???
     } 
@@ -137,7 +137,7 @@ object InmemoryBlobStorage {
 
 
 object InmemoryMetadataRepo {
-  val layer: ZLayer[Any, Nothing, MetadataRepo] = 
+  val layer = 
     ZLayer {
       ???
     }

